R/pseudocolor_plot.R

```diff
@@ -0,0 +1,346 @@
+
+
+#' Get density of points in 2 dimensions.
+#'
+#' Calculate the kernel density of points in 2 dimensions.
+#'
+#' @param x A numeric vector.
+#' @param y A numeric vector.
+#' @param n Create a square n by n grid to compute density.
+#' @param ... Additional arguments to pass to 'MASS::kde2d'.
+#' @return The density within each square.
+#'
+#' @noRd
+#'
+#' @examples
+#'
+#' library(pixelatorR)
+#'
+#' x <- rnorm(1000)
+#' y <- rnorm(1000)
+#'
+#' get_density(x, y, n = 100)
+#'
+.get2Ddensity <- function(
+    x,
+    y,
+    n = 500,
+    ...
+) {
+  # Check that MASS is installed
+  expect_MASS()
+
+  dens_grid <-
+    MASS::kde2d(x, y, n = n, ...)
+
+  dens <-
+    dens_grid$z[cbind(findInterval(x, dens_grid$x),
+                      findInterval(y, dens_grid$y))]
+
+  return(dens)
+
+}
+
+#' Create a pseudocolor plot.
+#'
+#' Create a pseudocolor plot of two markers.
+#'
+#' @param object A Seurat object.
+#' @param marker1 Marker to plot along the x-axis.
+#' @param marker2 Marker to plot along the y-axis.
+#' @param facet_vars Variables to facet the plot by.
+#' @param plot_gate A data.frame with columns 'xmin', 'xmax', 'ymin', 'ymax' to plot a gate. This data.frame can also
+#'                  contain the variables in 'facet_vars' to plot different gates by facets.
+#' @param scale_density Scale the density to the maximum density per facet.
```

Comment:
Nitpicking but maybe explicitly state that setting it to FALSE will scale across the whole dataset.